solana-bridges/solana-client/SolanaClient.sol

```diff
@@ -588,17 +590,36 @@ struct Slot {
     uint64 firstNormalSlot;
     uint64 slotsPerEpoch;
 
-
     uint64 public seenBlocks;
     bytes32 public lastHash;
     uint64 public lastSlot;
     Slot[HISTORY_SIZE] public slots;
 
-    event Success();
+    function getSlot(uint64 slot) public view returns (Slot memory) {
+        return slots[slotOffset(slot)];
+    }
 
-    constructor () public {
+    // Workarounds for misbehaving hs-web3 bindings
+    function getSlot_(uint64 slot) public view returns (bool, bytes32, bytes32, uint64) {
+        Slot storage s = slots[slotOffset(slot)];
+        return (s.hasBlock, s.blockHash, s.leaderPublicKey, s.voteCounts);
+    }
+    function getSignatures(uint64 slot, uint64 transactionIndex) public view returns (bytes memory) {
+        return getSlot(slot).transactionSignatures[transactionIndex];
+    }
+    function getMessage(uint64 slot, uint64 transactionIndex) public view returns (bytes memory) {
+        return getSlot(slot).transactionMessages[transactionIndex];
+    }
+
+
+    uint constant challengePayout = 10*1000*1000;
+    constructor () payable public {
         creator = msg.sender;
+        if(address(this).balance < challengePayout) {
```

Comment:
Eh? The contract needs the deposit to be done at some point - the way it's currently done the deposit is required on deployment since the check is done on the constructor.
The alternative would be to add a separate `payable` "deposit", and move this check to the endpoints receiving info from the relayer